src/fields/ntt.rs

```diff
@@ -0,0 +1,268 @@
+use crate::fields::FieldElement;
+
+/// Represents an element of an NTT-friendly field.
+///
+/// Fields implementing this trait must have a subgroup under multiplication with
+pub trait NttFieldElement: FieldElement {
+    /// A 2^k-th root of unity in the field.
+    ///
+    /// A 2^k-th root of unity of a field is a generator of the subgroup of the multiplicative group
+    /// with order 2^k. Thus, it satisfies omega^(2^k) = 1, and omega^i != 1 for 0 < i < 2^k.
+    const ROOT_OF_UNITY: Self;
+
+    /// A 2^k-th root of unity in the field, and the inverse of [`Self::ROOT_OF_UNITY`].
+    const ROOT_OF_UNITY_INVERSE: Self;
+
+    /// The base-2 logarithm of the order of `ROOT_OF_UNITY` in the multiplicative group.
+    const LOG2_ROOT_ORDER: usize;
+
+    /// The multiplicative inverse of 2 in this field.
+    const HALF: Self;
+
+    /// Computes the Number Theoretic Transform of a sequence. The result is returned in-place in
+    /// bit-reversed order.
+    ///
+    /// The `omegas` argument must be a list of power-of-two roots of unity, such that element i is
+    /// the 2^i-th root of unity. It should start with 1 itself, and contain at least enough values
+    /// to include the `values.len()`-th root of unity. Note that for each element in the array, its
+    /// predecessor is its square.
+    ///
+    /// # Panics
+    ///
+    /// This panics if the length of the input is not a power of two.
+    ///
+    /// This panics if there are not enough roots of unity in `omegas`.
+    fn ntt_bit_reversed(values: &mut [Self], omegas: &[Self]) {
+        let log_n = usize::try_from(values.len().ilog2()).unwrap();
+        if 1 << log_n != values.len() {
```

Comment:
We should decide as a matter of style whether we prefer `1 << n` or `2usize.pow(n)` and be consistent about it. I suspect the compiler emits shifts anyway, but I think I prefer `1 << n` because then we don't have to coerce the argument to `pow` into a `u32`. WDYT?

Reply:
I moved to `1 << n` in #85 for the reasons I argued above.